### DISCUSSION TITLE
Should methods that return txs include ignored txs by block-specific methods?

### DISCUSSION THREAD

**Discussion by acuarica** [hiero-ledger/hiero-json-rpc-relay#3218] [General]
Following up on issue https://github.com/hashgraph/hedera-json-rpc-relay/issues/3171, there are special kind of transactions, like those with `INVALID_ACCOUNT_ID` status, that are filtered out by methods like `eth_getBlockByNumber`. This is because those kind of transactions are Hedera-specific and don't have an Ethereum equivalent.

To provide a consistent experience, we need to decide whether or not these kind of transactions should be retrievable through other method calls. That is, should methods like `eth_getTransactionByHash`, return transactions **not** returned by block-specific methods?

### Related

- https://github.com/hashgraph/hedera-json-rpc-relay/discussions/567

**Comment by Nana-EC** [comment]
Following up on this question on whether we should also filter out transactions that failed to be executed by the EVM - I think for consistency with other EVM's the answer is yes.

Essentially an EVM client on Ethereum will only submit a transaction to consensus that successfully passes precheck.
On the other hand, Hedera will come to consensus on transaction where the submitter can pay the cost of the transactions, it doesn't care if precheck fails.
As a result we have the unique case EVM transactions that went through consensus but failed precheck on the CN and actually never went through the EVM.

I think the final check is whether the EOAs nonce was incremented.
We should only include transactions that moved the nonce counter for the EOA that submitted. 
Any transaction that failed check prior to nonce bump should be excluded from `eth_getTransactionReceipt` and `eth_getBlockByHash` and `eth_getBlockByHash`.

What do you think @acuaric, @natanasow and @lukelee-sl ?
